app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -26,10 +26,29 @@ def index
       respond(collection: collection)
     end
 
+    api :POST, "/organizations/:organization_id/upstream_subscriptions",
+      N_("Add subscriptions consumed by a manifest from Red Hat Subscription Management")
+    param :_json, Array, desc: N_("Array of pools to add"), required: true do
```

Comment:
It doesn't look we are using the :_json param anywhere else in our controllers which makes it unique - do we actually want to treat input to this API differently?

Reply:
Sorry, how else do we depict a body in apipie?

Reply:
This is probably the correct way to do that but is that what we actually want to do with this API instead of taking the input as a parameter within the body?

It seems inconsistent with the rest of the API and I'm basing that on the fact that we're not using :_json elsewhere. Maybe it *is* consistent and I'm just missing that in which case I could use some clarification. Sorry to raise this sort of question late in the review game, but it seems valid!

Reply:
> taking the input as a parameter within the body

Modifying this would change the API, which would upset @waldenraines. We had this conversation [here](https://github.com/Katello/katello/pull/7235#issuecomment-373759678). Changing the API we agreed on isn't really an option, imo.

Reply:
Yep, that convo did happen and the frontend would need to change - but would you say this approach is consistent or inconsistent with our API in general? I think that is the important thing here. We can worry about his feelings later, sorry walden!

Reply:
I agree that it's inconsistent with our API (I can't find any other places we accept a body without being nested in a param), and apipie doesn't seem to have a better way to support it other than using `:_json`, which also generates poor documentation. Still, I don't want to move forward without @waldenraines at least OKing it.